components/supersaas/supersaas.app.js

```diff
@@ -0,0 +1,66 @@
+const envConf = require('./envConf.js');
+
+module.exports = {
+  type: 'app',
+  app: 'supersaas',
+
+  propDefinitions: {
+    schedules: {
+      type: 'string[]',
+      label: 'Schedules',
+      description: `The schedules you'd like to watch for changes`,
+      async options() {
+        return await this.getSchedules();
+      },
+    },
+  },
+
+  methods: {
+    async axios(path, opts = {}) {
+      const { axios } = await require('@pipedreamhq/platform');
+
+      return await axios(this, {
+        url: `${envConf.urlPrefix}${path}`,
+
+        ...opts,
+
+        params: {
+          account: this.$auth.account,
+          api_key: this.$auth.api_key,
+
+          ...opts.params || {},
+        },
+      });
+    },
+
+    async getSchedules() {
+      const xs = await this.axios('/api/schedules.json');
+      return xs.map(x => ({ value: x.id, label: x.name }));
+    },
+
+    async createHooks(hookParams) {
+      const { axios } = this;
+
+      console.log('Creating hooks:', hookParams);
+
+      return await Promise.all(hookParams.map(
+        x => axios('/api/hooks', { method: 'POST', params: x }),
+      ));
+    },
+
+    // TODO: Better error handling. Dylan suggested retries with a backoff
```

Comment:
I'm not too worried about this for now, but we'll consider ways to improve this kind of error handling in the future as we talked about in Slack.